src/processes/helper/sendTopicAutoMessage.js

```diff
@@ -0,0 +1,61 @@
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const {sample} = require('lodash');
+const {followTopicQueue} = require('../../config/redis');
+const {CommunityMessageFormat, LogError} = require('../../databases/models');
+
+const followTopicServiceQueue = async (userId, topicId, communityMessageFormatId, delay) => {
+  const data = {
+    userId,
+    topicId,
+    communityMessageFormatId
+  };
+  const currentTime = momentTz().tz('America/Los_Angeles');
+  const randomTime = sample([6, 7, 8]);
+  const additionalDays = delay;
+
+  const requiredTime = momentTz()
+    .tz('America/Los_Angeles')
+    .set({hour: randomTime})
+    .add(additionalDays, 'days');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
+};
+const sendTopicAutoMessage = async (userId, topicIds) => {
+  try {
+    topicIds.forEach(async (topicId) => {
+      const communityMessageFormats = await CommunityMessageFormat.findAll({
+        where: {topic_id: topicId, status: '1'}
+      });
+
+      if (communityMessageFormats.length > 0) {
+        communityMessageFormats.forEach(async (communityMessageFormat) => {
+          followTopicServiceQueue(
+            userId,
+            topicId,
+            communityMessageFormat.community_message_format_id,
+            communityMessageFormat.delay
+          );
+        });
+      }
+    });
+  } catch (error) {
+    await LogError.create({
+      message: error.message
+    });
+    console.error('Register send Topic auto message: ', error);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `sendTopicAutoMessage` function, you're using `forEach` with async functions. However, `forEach` will not wait for promises to resolve before moving on to the next iteration. This could lead to unexpected behavior if the order of execution matters in your code. Consider using `for...of` or `Promise.all` instead.

```diff
- topicIds.forEach(async (topicId) => {
+ for (const topicId of topicIds) {
      const communityMessageFormats = await CommunityMessageFormat.findAll({
        where: {topic_id: topicId, status: '1'}
      });

      if (communityMessageFormats.length > 0) {
-       communityMessageFormats.forEach(async (communityMessageFormat) => {
+       for (const communityMessageFormat of communityMessageFormats) {
          followTopicServiceQueue(
            userId,
            topicId,
            communityMessageFormat.community_message_format_id,
            communityMessageFormat.delay
          );
-       });
+       }
      }
- });
+ }
```
Please note that if the order of execution doesn't matter and you want to execute all promises concurrently, you can use `Promise.all`:

```diff
- topicIds.forEach(async (topicId) => {
+ await Promise.all(topicIds.map(async (topicId) => {
      const communityMessageFormats = await CommunityMessageFormat.findAll({
        where: {topic_id: topicId, status: '1'}
      });

      if (communityMessageFormats.length > 0) {
-       communityMessageFormats.forEach(async (communityMessageFormat) => {
+       await Promise.all(communityMessageFormats.map(async (communityMessageFormat) => {
          followTopicServiceQueue(
            userId,
            topicId,
            communityMessageFormat.community_message_format_id,
            communityMessageFormat.delay
          );
-       });
+       }));
      }
- });
+ }));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->